test/byContract/IexecPoco/06_claim.test.ts

```diff
@@ -307,6 +306,126 @@ describe('Poco', async () => {
         await expect(iexecPoco.connect(anyone).claim(taskId)).to.be.reverted;
     });
 
+    describe('Claim array', function () {
+        it('Should claim array', async function () {
+            const volume = 3;
+            const { orders } = buildOrders({
+                assets: ordersAssets,
+                requester: requester.address,
+                prices: ordersPrices,
+                volume,
+            });
+            const { dealId, startTime } = await signAndMatchOrders(orders);
+            const taskIds = [];
+            for (let taskIndex = 0; taskIndex < volume; taskIndex++) {
+                taskIds.push(getTaskId(dealId, taskIndex));
+                await iexecPoco
+                    .connect(scheduler)
+                    .initialize(dealId, taskIndex)
+                    .then((tx) => tx.wait());
+            }
+            // Mine empty block so timestamp is accurate when static call is made
+            await time.setNextBlockTimestamp(startTime + maxDealDuration).then(() => mine());
+
+            expect(await iexecPoco.connect(anyone).callStatic.claimArray(taskIds)).to.be.true;
+            const claimArrayTx = await iexecPoco.connect(anyone).claimArray(taskIds);
+            await claimArrayTx.wait();
+            for (let taskId of taskIds) {
+                await expect(claimArrayTx).to.emit(iexecPoco, 'TaskClaimed').withArgs(taskId);
+            }
+        });
+
+        it('Should not claim array when one is not claimable', async function () {
+            const volume = 2;
+            const { orders } = buildOrders({
+                assets: ordersAssets,
+                requester: requester.address,
+                prices: ordersPrices,
+                volume,
+            });
+            const { dealId, startTime } = await signAndMatchOrders(orders);
+            const taskIndex1 = 0;
+            const taskIndex2 = 1;
+            const taskId1 = getTaskId(dealId, taskIndex1);
+            const taskId2 = getTaskId(dealId, taskIndex2);
+            await iexecPoco
+                .connect(scheduler)
+                .initialize(dealId, taskIndex1)
+                .then((tx) => tx.wait());
+            await time.setNextBlockTimestamp(startTime + maxDealDuration);
```

Comment:
Do we need to mine or it's only for static calls?

Reply:
Only for static yes :ok_hand: